app/src/main/java/com/evelynhasama/bucketbuds/MusementHelper.java

```diff
@@ -0,0 +1,77 @@
+package com.evelynhasama.bucketbuds;
+
+import android.content.Context;
+import android.util.Log;
+import com.android.volley.Response;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.text.ParseException;
+
+public class MusementHelper implements IEventAPI {
+    public static final String TAG = "MusementHelper";
+    public static final String BASE_URL = "https://sandbox.musement.com/api/v3/activities?";
+    public static final String PARAM_COORDS = "coordinates=";
+    public static final String PARAM_DISTANCE = "&distance=";
+    public static final String PARAM_LIMIT="&limit=10";
+    public static final String UNIT="M";
+    private static MusementHelper mInstance;
+
+    private MusementHelper(){
+    }
+
+    public static synchronized MusementHelper getInstance() {
+        if (mInstance == null) {
+            mInstance = new MusementHelper();
+        }
+        return mInstance;
+    }
+
+    public void getEvents(Context context, Double latitude, Double longitude, InspoActivitiesAdapter adapter, String radiusFilter) {
+
+        String coords = latitude + "," + longitude;
+        String url = BASE_URL + PARAM_COORDS + coords + PARAM_LIMIT + PARAM_DISTANCE + radiusFilter + UNIT;
+
+        Response.Listener<String> responseListener = ApiHelper.buildResponseListener(
+            (responseObject -> {
+                try {
+                    return responseObject.getJSONArray("data");
```

Comment:
nice! Since the `try{}catch(){}` happens everywhere you could even take care of that in the `buildResponseListener` method

Reply:
yeah there was a lot of try/catch, I'll try that!